<a href="https://colab.research.google.com/github/H2D2R2/Dz2/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Задание 1
Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [ ]:
# Решаем методом Биномиального распределения

from math import factorial    

def combinations(n, k):                                                         
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

p = 0.8 
n = 100
k = 85
q = 1 - p

combinations(n,k) * (p ** k) * (q ** (n-k)) * 100                               #загружаем данные в формулу

4.806179370074635

**Ответ:** вероятность, что стрелок попадет в цель ровно 85 раз - 4.8%

# Задание 2
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [ ]:
# Решаем методом Распределение Пуассона

p = 0.0004 #вероятность
n = 5000   #число лампочек
m1 = 0     #количество перегоревших для первого условия
m2 = 2     #количество перегоревших для второго условия
lambda_ = p * n
lambda_2 = - (p * n )
e = 2.72

(lambda_ ** m1) / factorial(m1) * (e ** lambda_2) *100                          #загружаем данные в формулу

13.516435986159166

In [ ]:
(lambda_ ** m2) / factorial(m2) * (e ** lambda_2) *100                          #загружаем данные для второго условия в формулу 

27.032871972318333

**Ответ:** Вероятность отсутствия перегоревших лампочек - 13,5%. Вероятность, что перегорят ровно две - 27%

# Задание 3
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [ ]:
# Решаем методом Биномиального распределения

def combinations(n, k):                                                         
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

p = 0.5 
n = 144
k = 70
q = 1 - p

combinations(n,k) * (p ** k) * (q ** (n-k)) * 100                               #загружаем данные в формулу 

6.281178035144776

**Ответ:** вероятность выпадения орла 70 раз - 6.28 %.

# Задание 4

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [ ]:
(7/10 * 6/9) + (9/11 * 8/10)

1.121212121212121

In [ ]:
(7/10 * 7/9) + (9/11 * 9/10)

1.2808080808080808

**Ответ:** вероятность вытащить все белые мечи такая же, как и достать белым хотя бы один мяч - 1.21 . Вероятность вытащить ровно два белых мяча - 1.28 .